ztp/gitops-subscriptions/argocd/README.md

```diff
@@ -49,14 +49,68 @@ These steps configure your hub cluster with a set of ArgoCD Applications which g
-- Add the SiteConfig CR for your site to your git repository
-    Note: The extra-manifest Machine configs exist under [source-crs/extra-manifest](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/source-crs/extra-manifest) will be included in the generated configMap for extra manifest.
-    Optional: For adding other extra-manifests to the provisioned cluster, create a directory ex; `sno-extra-manifest/` in relative path at siteconfig GIT repository and add the extra-manifest files to it. Then in the SiteConfig.yaml  set the extraManifestPath field ex; `extraManifestPath: sno-extra-manifest/` .
-- Add the PolicyGenTemplate CR for your site to your git repository
-- Push your changes to the git repository. The SiteConfig and PolicyGenTemplate CRs may be pushed simultaneously.
+1. Create the required secrets for site. These resources must be in a namespace with a name matching the cluster name. In out/argocd/example/siteconfig/example-sno.yaml the cluster name & namespace is `example-sno`
+   1. Create the namespace for the cluster:
+```
+$ export CLUSTERNS=example-sno
+$ oc create namespace $CLUSTERNS
+```
+   2. Create a pull secret for the cluster. The pull secret must contain all credentials necessary for installing OpenShift and all required operators. In all of the example SiteConfigs this is named `assisted-deployment-pull-secret`
+```
+$ oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: assisted-deployment-pull-secret
+  namespace: $CLUSTERNS
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: $(base64 <pull-secret.json)
+EOF
+```
+   3. Create a BMC authentication secret for each host you will be deploying.
+```
+$ oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: $(read -p 'Hostname: ' tmp; echo -n $tmp)-bmc-secret
+  namespace: $CLUSTERNS
+type: Opaque
+data:
+  username: $(read -p 'Username: ' tmp; base64 <<<$tmp)
+  password: $(read -s -p 'Password: ' tmp; base64 <<<$tmp)
+EOF
+```
+2. Create a SiteConfig CR for your cluster in your local clone of the git repository:
+   1. Begin by choosing an appropriate example from out/argocd/example/siteconfig/.  There are examples there for SNO, 3-node, and standard clusters.
+   2. Change the cluster and host details in the example to match your desired cluster.  Some important notes:
+      - The clusterImageSetNameRef must match an imageset available on the hub cluster (run `oc get clusterimagesets` for the list of supported versions on your hub)
+      - Ensure the cluster networking sections are defined correctly:
+        - For SNO deployments, you must define a `MachineNetwork` section and not the `apiVIP` and `ingressVIP` values.
+        - For 3-node and standard deployments, you must define the `apiVIP` and `ingressVIP` values and not the `MachineNetwork` section.
```

Comment:
We should see if there is an effort underway to rationalize these so that specification of the api/ingress is consistent.